pkg/api/validation/apisix_plugin_config_test.go

```diff
@@ -0,0 +1,94 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"testing"
+
+	v2 "github.com/apache/apisix-ingress-controller/pkg/kube/apisix/apis/config/v2"
+)
+
+func TestValidateApisixRoutePlugins(t *testing.T) {
+	validPlugins := []v2.ApisixRoutePlugin{
+		{
+			Name: "plugin1",
+			Config: v2.ApisixRoutePluginConfig{
+				"param1": "value1",
+				"param2": 123,
+			},
+		},
+		{
+			Name: "plugin2",
+			Config: v2.ApisixRoutePluginConfig{
+				"param1": "value2",
+			},
+		},
+	}
+
+	invalidPlugins := []v2.ApisixRoutePlugin{
+		{
+			Name:   "",
+			Config: v2.ApisixRoutePluginConfig{},
+		},
+		{
+			Name: "plugin1",
+			Config: v2.ApisixRoutePluginConfig{
+				"param1": "value1",
+				"param2": "invalid",
+			},
+		},
+	}
+
+	tests := []struct {
```

Comment:
You may use the same test cases. I am not sure but @Gallardot can tell this more.
https://github.com/apache/apisix-ingress-controller/blob/2182a48cbca785373eca745a13d8cf2b7d9ab6c8/pkg/api/validation/apisix_route_test.go#L73-L76

Reply:
If I add `ValidatePlugin` then will have to update the test cases. @Gallardot Kindly review.

Reply:
Let's wait for it to get reviewed. TY